Sources/Amplitude/Amplitude.m

```diff
@@ -1059,6 +1066,18 @@ - (void)makeEventUploadPostRequest:(NSString *)url events:(NSString *)events num
                 AMPLITUDE_LOG(@"Request too large, will decrease size and attempt to reupload");
                 self->_updatingCurrently = NO;
                 [self uploadEventsWithLimit:self->_backoffUploadBatchSize];
+            } else if ([httpResponse statusCode] == 429) {
+                // if rate limited
+                self->_backoffUpload = YES;
+                if (self->_numRetries == 0) {
+                    // cache the recent valus of event upload period
+                    self->_originalUploadPeriosdInSeconds = self.eventUploadPeriodSeconds;
```

Comment:
Should we somehow reset this to the original value after we successfully upload? I think now this will only keep increasing but never returns to the original upload period right?

Reply:
I reset the eventUploadPeriodSeconds after upload success. The value cached here won't be updated each time.